### ISSUE TITLE
JsonResult with JsonSerializerSettings throws exception in .NET Core 3.1 / Azure Functions v3

### ISSUE THREAD

**Issue by readnikhil** [Azure/azure-functions-host#7102]
Hi All , 
I refered to this issue : https://github.com/Azure/azure-functions-host/issues/5376

But I still cannot solve it . My Azure function works fine locally but it gives following error when I deploy on Azure Portal using Azure pipeline . 
Method not found: 'Void Microsoft.AspNetCore.Mvc.JsonResult..ctor

Function App version: 3.0
Function App runtime: .Net Core 3.1

Hosted info:
Runtime: 3.0.15193.0 (~3)
Location: West Europe
Platform: App Service Plan (Y1: 0)
OS: Windows

I have added reference to nuget Microsoft.AspNetCore.Mvc.NewtonsoftJson and set FUNCTIONS_V2_COMPATIBILITY_MODE as true. 
This was working fine when I used Azure function v2 and .net core 2.1 .But stopped working when I upgraded it.
Can you please help what can be wrong ?

With Regards,
Nikhil Rai

**Comment by readnikhil** [comment]
Hi , Can someone please help with this ?

**Comment by v-bbalaiagar** [comment]
Hi @readnikhil, Apologies for the delayed response, can we know if you are still seeing this issue.

**Comment by readnikhil** [comment]
Hi , Yes it is still an issue . 
So it does not work with this code .
private static readonly JsonSerializerSettings _jsonSerializerSettings = new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore };

return new JsonResult(objectofclass , _jsonSerializerSettings);

This has worked before but suddently stopped working in all the solutions when they were redeployed .

**Comment by v-bbalaiagar** [comment]
Hi @readnikhil, Thank you for your feedback! We will check for the possibilities internally and update you with the findings.

**Comment by v-bbalaiagar** [comment]
Hi @readnikhil, Can we know what version of package you are using.

**Comment by readnikhil** [comment]
[image: image.png]

**Comment by v-bbalaiagar** [comment]
Hi @readnikhil, Kindly can you share it again. It looks like image is not loaded.

**Comment by readnikhil** [comment]
![image](https://user-images.githubusercontent.com/50196155/131086752-95857b61-1f79-4182-96aa-3f0931b3f2fb.png)

Please let me know if you can see the image now . 
MVC.NewtonsoftJson(3.1.11)
Sdk.Functions(3.0.11)

**Comment by v-bbalaiagar** [comment]
Yes, I can see the image now. We will investigate this further and update as appropriate.

**Comment by v-bbalaiagar** [comment]
Hi @readnikhil, Apologies for the delayed response. Discussed this internally, let us know if you are still facing this issue. If yes, Can you give us the repro steps to understand the process that you are following.

**Comment by readnikhil** [comment]
Hi , Yes the issue still persists . I have a http trigger Azure function that converts a json message to another json format .
I need to restrict all null values elements so that they are not sent in outgoing json message as response to http trigger .
So I had used the nuget packages that I have mentioned and then we try to set this property .
![image](https://user-images.githubusercontent.com/50196155/134617634-e8c404ff-8c86-432e-9c63-bd777a890dfc.png)

After converting message , we try to use this property and return the result.
![image](https://user-images.githubusercontent.com/50196155/134617737-4dc78df0-057d-4c42-b420-d092fcb86961.png)

Both source and destination class objects are marked with Jsonpropery.
![image](https://user-images.githubusercontent.com/50196155/134617840-c804cffa-83a8-45c9-acda-60c787f1aee9.png)

Please let me know if you need any other details .

With Regards,
Nikhil Rai

**Comment by v-bbalaiagar** [comment]
Tagging @fabiocav for further investigation

**Comment by fabiocav** [comment]
@readnikhil we're flagging this for investigation, but if you could have a minimal repro in a repo here, that would ensure we're working with the same configuration you're working with.
